include/FockSpace/BaseFockSpace.hpp

```diff
@@ -10,6 +10,15 @@
 namespace GQCG {
 
 
+/**
+ *  An enum class for the implemented Fock space types
+ */
+enum class FockSpaceType {
```

Comment:
I think it would be better if this would be inside a separate header `FockSpaceType.hpp`.